storage: TestIntentInterleavingIter failed

[(storage).TestIntentInterleavingIter failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2560362&tab=buildLog) on [master@ced6751a28a1803783ed30151b39caeefc45147b](https://github.com/cockroachdb/cockroach/commits/ced6751a28a1803783ed30151b39caeefc45147b):

```
=== RUN   TestIntentInterleavingIter
--- FAIL: TestIntentInterleavingIter (0.05s)
=== RUN   TestIntentInterleavingIter/error_race
    intent_interleaving_iter_test.go:185: 
        testdata/intent_interleaving_iter/error_race:19: seek-ge k=a
        next
        next
        next
        next
        next
        seek-lt k=d
        prev
        prev
        prev
        prev
        output didn't match expected:
        @@ -1,12 +1,12 @@
        -seek-ge "a"/0,0: output: meta k=a ts=10 txn=1
        -next: output: value k=a ts=10 v=a10
        -next: output: meta k=b ts=20 txn=4
        +seek-ge "a"/0,0: output: meta k=a ts=10.000000000,0 txn=1
        +next: output: value k=a ts=10.000000000,0 v=a10
        +next: output: meta k=b ts=20.000000000,0 txn=4
         next: output: err: intentIter incorrectly positioned, cmp: 0
         next: output: err: intentIter incorrectly positioned, cmp: 0
         next: output: err: intentIter incorrectly positioned, cmp: 0
        -seek-lt "d"/0,0: output: value k=c ts=30 v=c30
        -prev: output: meta k=c ts=30 txn=4
        -prev: output: value k=b ts=20 v=b20
        -prev: output: meta k=b ts=20 txn=2
        +seek-lt "d"/0,0: output: value k=c ts=30.000000000,0 v=c30
        +prev: output: meta k=c ts=30.000000000,0 txn=4
        +prev: output: value k=b ts=20.000000000,0 v=b20
        +prev: output: meta k=b ts=20.000000000,0 txn=2
         prev: output: err: intentIter should not be after iter
         
    --- FAIL: TestIntentInterleavingIter/error_race (0.01s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestIntentInterleavingIter PKG=./pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestIntentInterleavingIter.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(storage).TestIntentInterleavingIter failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2560442&tab=buildLog) on [master@b2ac447d23c97332b705e288c055d63119fb8d28](https://github.com/cockroachdb/cockroach/commits/b2ac447d23c97332b705e288c055d63119fb8d28):

```
=== RUN   TestIntentInterleavingIter
--- FAIL: TestIntentInterleavingIter (0.05s)
=== RUN   TestIntentInterleavingIter/error_race
    intent_interleaving_iter_test.go:185: 
        testdata/intent_interleaving_iter/error_race:19: seek-ge k=a
        next
        next
        next
        next
        next
        seek-lt k=d
        prev
        prev
        prev
        prev
        output didn't match expected:
        @@ -1,12 +1,12 @@
        -seek-ge "a"/0,0: output: meta k=a ts=10 txn=1
        -next: output: value k=a ts=10 v=a10
        -next: output: meta k=b ts=20 txn=4
        +seek-ge "a"/0,0: output: meta k=a ts=10.000000000,0 txn=1
        +next: output: value k=a ts=10.000000000,0 v=a10
        +next: output: meta k=b ts=20.000000000,0 txn=4
         next: output: err: intentIter incorrectly positioned, cmp: 0
         next: output: err: intentIter incorrectly positioned, cmp: 0
         next: output: err: intentIter incorrectly positioned, cmp: 0
        -seek-lt "d"/0,0: output: value k=c ts=30 v=c30
        -prev: output: meta k=c ts=30 txn=4
        -prev: output: value k=b ts=20 v=b20
        -prev: output: meta k=b ts=20 txn=2
        +seek-lt "d"/0,0: output: value k=c ts=30.000000000,0 v=c30
        +prev: output: meta k=c ts=30.000000000,0 txn=4
        +prev: output: value k=b ts=20.000000000,0 v=b20
        +prev: output: meta k=b ts=20.000000000,0 txn=2
         prev: output: err: intentIter should not be after iter
         
    --- FAIL: TestIntentInterleavingIter/error_race (0.01s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestIntentInterleavingIter PKG=./pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestIntentInterleavingIter.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(storage).TestIntentInterleavingIter failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2560509&tab=buildLog) on [master@04063a9f699d37dd95696210a52e5c6d8f536b86](https://github.com/cockroachdb/cockroach/commits/04063a9f699d37dd95696210a52e5c6d8f536b86):

```
=== RUN   TestIntentInterleavingIter
--- FAIL: TestIntentInterleavingIter (0.05s)
=== RUN   TestIntentInterleavingIter/error_race
    intent_interleaving_iter_test.go:185: 
        testdata/intent_interleaving_iter/error_race:19: seek-ge k=a
        next
        next
        next
        next
        next
        seek-lt k=d
        prev
        prev
        prev
        prev
        output didn't match expected:
        @@ -1,12 +1,12 @@
        -seek-ge "a"/0,0: output: meta k=a ts=10 txn=1
        -next: output: value k=a ts=10 v=a10
        -next: output: meta k=b ts=20 txn=4
        +seek-ge "a"/0,0: output: meta k=a ts=10.000000000,0 txn=1
        +next: output: value k=a ts=10.000000000,0 v=a10
        +next: output: meta k=b ts=20.000000000,0 txn=4
         next: output: err: intentIter incorrectly positioned, cmp: 0
         next: output: err: intentIter incorrectly positioned, cmp: 0
         next: output: err: intentIter incorrectly positioned, cmp: 0
        -seek-lt "d"/0,0: output: value k=c ts=30 v=c30
        -prev: output: meta k=c ts=30 txn=4
        -prev: output: value k=b ts=20 v=b20
        -prev: output: meta k=b ts=20 txn=2
        +seek-lt "d"/0,0: output: value k=c ts=30.000000000,0 v=c30
        +prev: output: meta k=c ts=30.000000000,0 txn=4
        +prev: output: value k=b ts=20.000000000,0 v=b20
        +prev: output: meta k=b ts=20.000000000,0 txn=2
         prev: output: err: intentIter should not be after iter
         
    --- FAIL: TestIntentInterleavingIter/error_race (0.01s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestIntentInterleavingIter PKG=./pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestIntentInterleavingIter.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(storage).TestIntentInterleavingIter failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2560595&tab=buildLog) on [master@09ccfed78308f45abaf09cc993c82a18f6719f85](https://github.com/cockroachdb/cockroach/commits/09ccfed78308f45abaf09cc993c82a18f6719f85):

```
=== RUN   TestIntentInterleavingIter
--- FAIL: TestIntentInterleavingIter (0.05s)
=== RUN   TestIntentInterleavingIter/error_race
    intent_interleaving_iter_test.go:185: 
        testdata/intent_interleaving_iter/error_race:19: seek-ge k=a
        next
        next
        next
        next
        next
        seek-lt k=d
        prev
        prev
        prev
        prev
        output didn't match expected:
        @@ -1,12 +1,12 @@
        -seek-ge "a"/0,0: output: meta k=a ts=10 txn=1
        -next: output: value k=a ts=10 v=a10
        -next: output: meta k=b ts=20 txn=4
        +seek-ge "a"/0,0: output: meta k=a ts=10.000000000,0 txn=1
        +next: output: value k=a ts=10.000000000,0 v=a10
        +next: output: meta k=b ts=20.000000000,0 txn=4
         next: output: err: intentIter incorrectly positioned, cmp: 0
         next: output: err: intentIter incorrectly positioned, cmp: 0
         next: output: err: intentIter incorrectly positioned, cmp: 0
        -seek-lt "d"/0,0: output: value k=c ts=30 v=c30
        -prev: output: meta k=c ts=30 txn=4
        -prev: output: value k=b ts=20 v=b20
        -prev: output: meta k=b ts=20 txn=2
        +seek-lt "d"/0,0: output: value k=c ts=30.000000000,0 v=c30
        +prev: output: meta k=c ts=30.000000000,0 txn=4
        +prev: output: value k=b ts=20.000000000,0 v=b20
        +prev: output: meta k=b ts=20.000000000,0 txn=2
         prev: output: err: intentIter should not be after iter
         
    --- FAIL: TestIntentInterleavingIter/error_race (0.01s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestIntentInterleavingIter PKG=./pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestIntentInterleavingIter.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/storage/testdata/intent_interleaving_iter/error_race
+++ b/pkg/storage/testdata/intent_interleaving_iter/error_race
@@ -29,14 +29,14 @@ prev
 prev
 prev
 ----
-seek-ge "a"/0,0: output: meta k=a ts=10 txn=1
-next: output: value k=a ts=10 v=a10
-next: output: meta k=b ts=20 txn=4
+seek-ge "a"/0,0: output: meta k=a ts=10.000000000,0 txn=1
+next: output: value k=a ts=10.000000000,0 v=a10
+next: output: meta k=b ts=20.000000000,0 txn=4
 next: output: err: intentIter incorrectly positioned, cmp: 0
 next: output: err: intentIter incorrectly positioned, cmp: 0
 next: output: err: intentIter incorrectly positioned, cmp: 0
-seek-lt "d"/0,0: output: value k=c ts=30 v=c30
-prev: output: meta k=c ts=30 txn=4
-prev: output: value k=b ts=20 v=b20
-prev: output: meta k=b ts=20 txn=2
+seek-lt "d"/0,0: output: value k=c ts=30.000000000,0 v=c30
+prev: output: meta k=c ts=30.000000000,0 txn=4
+prev: output: value k=b ts=20.000000000,0 v=b20
+prev: output: meta k=b ts=20.000000000,0 txn=2
 prev: output: err: intentIter should not be after iter

